spec/requests/admin/connected_apps_controller_spec.rb

```diff
@@ -111,6 +111,38 @@
         end
       end
 
+      context "when VINE API is not set up properly" do
+        before do
+          # VineApiService will raise a KeyError if VINE_API_URL is not set
+          allow(VineApiService).to receive(:new).and_raise(KeyError)
```

Comment:
I think it would be better set up the conditions for this error to occur. Eg what if, in a future refactor, the `KeyError` is raised from another class.

IE does this do the trick?
```suggestion
          allow(ENV).to receive(:fetch).with("VINE_API_URL").and_return(nil)
```

Reply:
But if it's not that simple, then I approve this 👍

Reply:
> Eg what if, in a future refactor, the KeyError is raised from another class.

That's a good point, it doesn't work with `and_return(nil)` it would result in a different error but I found a way to trigger the error without relying on `VineApiService` raising the error.